packages/gatsby/cache-dir/app.js

```diff
@@ -19,6 +20,9 @@ setConfig({
   pureRender: true,
 })
 
+const preferDefault = m => (m && m.default) || m
+let Root = hot(preferDefault(require(`./root`)))
```

Comment:
So - as I mentioned in another comment - `hot` here would setup __current__ module as self-accepted, wiping and updating __everything__ _below_ it every time.
- react-dom
- socketIo
- emitter

It would be better to move it to `./root` or have another file, imported above this one with some important imports. That would link "important" modules with another "parent branch", and protect them during hot update.

Sound recommendation - have `hot` below(in module terms) `react-dom/render` and `redux/createStore` to maintain their values on update. You have to stop event propagation before it bubbles to the places you should keep between updates.

Reply:
@theKashey thanks again, I now moved it into `.cache/sync-requires.js` so only page components are hot loaded.

@DSchau This is not the most beautiful solution, but it was the best I could think of without applying `hot` further up. An alternative could be to map over `syncRequires` in `app.js` and call `hot` there instead of writing it into the generated `sync-requires.js`.